File: src/MyBlazorApp.Api.Contracts/WeatherForecastApiV1.cs
using System.Threading;
using System.Threading.Tasks;
using MyBlazorApp.Domain.WeatherForecasts;

namespace MyBlazorApp.Api.Contracts
{
    public static class WeatherForecastApiV1
    {
        public const string Version = "v1";
        public static class Routes
        {
            public const string Controller = Version + "/" + "weatherforecasts";
            public const string GetForecasts = "";
            public const string GetSuperForecasts = "superforecasts";
        }

        public static class Endpoints
        {
            public const string GetForecasts = Routes.Controller + "/" + Routes.GetForecasts;
            public const string GetSuperForecasts = Routes.Controller + "/" + Routes.GetSuperForecasts + "/";
        }

        public interface Interface
        {
            public Task<WeatherForecast[]> GetForecasts(int count, CancellationToken cancellationToken = default);
            public Task<WeatherForecast[]> GetSuperForecasts(int count, int superNumber, CancellationToken cancellationToken = default);
        }
    }
}

File: src/MyBlazorApp.Api.Contracts/WeatherForecastApiV1.fs
module MyBlazorApp.Api.Contracts.WeatherForecastApiV1

open System.Threading.Tasks
open MyBlazorApp.Domain.WeatherForecastsService

let [<Literal>] Version = "v1"

module Routes =
    let [<Literal>] Controller = "weather-forecasts" + "/" + Version + "/"
    let [<Literal>] GetForecasts = "forecasts"
    let [<Literal>] GetSuperForecasts = "super-forecasts"

module Endpoints =
    let [<Literal>] GetForecasts = Routes.Controller + Routes.GetForecasts
    let [<Literal>] GetSuperForecasts = Routes.Controller + Routes.GetSuperForecasts

type IService =
    abstract GetForecasts: count: int -> Task<WeatherForecast[]>
    abstract GetSuperForecasts: count: int -> superNumber: int -> Task<WeatherForecast[]>